### PR TITLE
docs(parameters): snippets split, improved, and lint

### LINKED ISSUE #1368: Refactor Parameters docs code snippets

### PR DESCRIPTION
**Pull request by leandrodamascena**
**Issue number:** #1368 

### Changes

Changes snippet fictitious names to real code snippet filenames for easier discovery and collaboration to improve/correct document snippets.

Some examples have also been refactored so that the codes are complete and there are no lint/pre-commit hooks issues.

### User experience

Before:
image

After:
image

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

-----
[View rendered docs/utilities/parameters.md](https://github.com/leandrodamascena/aws-lambda-powertools-python/blob/chore/parameters/docs/utilities/parameters.md)

### CODE REVIEW

File: docs/utilities/parameters.md [hunk 1]
```diff
@@ -37,7 +40,7 @@ This utility requires additional permissions to work as expected.
 
 ### Fetching parameters
 
-You can retrieve a single parameter  using `get_parameter` high-level function.
+You can retrieve a single parameter  using the `get_parameter` high-level function.
```
Comment (rubenfonseca) [inline docs/utilities/parameters.md]:
```suggestion
You can retrieve a single parameter using the `get_parameter` high-level function.
```

File: docs/utilities/parameters.md [hunk 2]
```diff
@@ -123,93 +126,95 @@ def handler(event, context):
 
 You can fetch secrets stored in Secrets Manager using `get_secrets`.
```
Comment (rubenfonseca) [inline docs/utilities/parameters.md]:
```suggestion
You can fetch secrets stored in Secrets Manager using `get_secret`.
```

File: docs/utilities/parameters.md [hunk 3]
```diff
@@ -334,94 +313,50 @@ DynamoDB provider can be customized at initialization to match your table struct
...
-from aws_lambda_powertools.utilities import parameters
-
-dynamodb_provider = parameters.DynamoDBProvider(
-	table_name="my-table",
-	key_attr="MyKeyAttr",
-	sort_attr="MySortAttr",
-	value_attr="MyvalueAttr"
-)
+=== "builtin_provider_dynamodb_custom_fields.py"
+    ```python hl_lines="3 5 9 10 15"
```
Comment (rubenfonseca) [inline docs/utilities/parameters.md]: On the rendered docs the highlight seems wrong. Can you please review it?

**Review by rubenfonseca — CHANGES_REQUESTED** [review]
This is amazing work @leandrodamascena thank you! I left some small comments but I ask you the following two things:

- can you please render the docs locally and manually check all the highlights to see if they make sense?
- can you rebase the branch with `develop`?

We should be able to merge this very soon

File: docs/utilities/parameters.md (hunk 3 above)
Comment (leandrodamascena) [reply docs/utilities/parameters.md]: Yep! Fixed.

File: docs/utilities/parameters.md (hunk 2 above)
Comment (leandrodamascena) [reply docs/utilities/parameters.md]: Done

**Comment by leandrodamascena** [comment]
> This is amazing work @leandrodamascena thank you! I left some small comments but I ask you the following two things:
> 
> * can you please render the docs locally and manually check all the highlights to see if they make sense?
> * can you rebase the branch with `develop`?
> 
> We should be able to merge this very soon

Hei Ruben! Thanks a lot for the feedback. I addressed all issues you mentioned and I moved 2 code snippet from `parameters.md` file to example folder.

**Review by rubenfonseca — APPROVED** [review]
Great work!

**Pull request merged**